src/Features/CSharp/Portable/Structure/CSharpStructureHelpers.cs

```diff
@@ -354,6 +354,7 @@ static SyntaxToken GetEndToken(SyntaxNode node)
                 }
                 else if (node.IsKind(SyntaxKind.ClassDeclaration, out TypeDeclarationSyntax typeDeclaration)
                     || node.IsKind(SyntaxKind.RecordDeclaration, out typeDeclaration)
+                    || node.IsKind(SyntaxKind.RecordStructDeclaration, out typeDeclaration)
                     || node.IsKind(SyntaxKind.StructDeclaration, out typeDeclaration)
                     || node.IsKind(SyntaxKind.InterfaceDeclaration, out typeDeclaration))
```

Comment:
```suggestion
                else if (node is TypeDeclarationSyntax typeDeclaration)
```

Not sure why we have so many patterns to express things that are trivial language checks :)